File: frontend/src/models/Exam.js
export class ExamResponse {
  constructor({ id, examName, startDate, endDate, examStatus, description }) {
    this.id = id
    this.examName = examName
    this.startDate = startDate
    this.endDate = endDate
    this.examStatus = examStatus
    this.description = description
  }

  static fromJSON(json) {
    return new ExamResponse(json)
  }
}

export class Exam {
  constructor(data) {
    this.id = data?.id ?? null;
    this.examCode = data?.examCode ?? null;
    this.examName = data?.examName ?? null;
    this.startDate = data?.startDate ?? null;
    this.endDate = data?.endDate ?? null;
    this.examStatus = data?.examStatus ?? null;
    this.description = data?.description ?? null;
    this.isOpen = data?.isOpen ?? false;
  }

  static fromJSON(json) {
    return new Exam(json)
  }
}

File: backend/src/main/java/com/examreg/examreg/models/Exam.java
package com.examreg.examreg.models;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import lombok.Data;

@Data
@Entity
public class Exam {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String examCode;

  private String examName;

  private LocalDate startDate;

  private LocalDate endDate;

  private String examStatus;

  @Column(columnDefinition = "TEXT")
  private String description; 

  private boolean isOpen;

  @ManyToMany
  @JoinTable(
    name = "exam_subject",
    joinColumns = @JoinColumn(name = "exam_id"),
    inverseJoinColumns = @JoinColumn(name = "subject_id")
  )
  private List<Subject> subjects;
}